### PR TITLE
Support non-EC2 targets with ob deploy

### PR DESCRIPTION
**Pull request by 3noch**
I have:

  - [x] Based work on latest `develop` branch
  - [x] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [x] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

**Comment by 3noch** [comment]
`module` is a really bad name for this...

File: default.nix [hunk 1]
```diff
@@ -167,18 +167,21 @@ in rec {
   '';
 
   serverModules = {
-    mkBaseEc2 = { hostName, routeHost, enableHttps, adminEmail, ... }: {...}: {
+    mkBaseEc2 = { obeliskPkgs, ... }: {...}: {
```
Comment (3noch) [inline default.nix]: Not a fan of this name/scheme either.

File: lib/command/src/Obelisk/Command/Deploy.hs [hunk 2]
```diff
@@ -72,6 +72,12 @@ deployInit thunkPtr deployDir sshKeyPath hostnames route adminEmail enableHttps
   --accidentally create a production deployment that uses development
   --credentials to connect to some resources.  This could result in, e.g.,
   --production data backed up to a dev environment.
+  putLog Notice "Writing default module.nix"
+  writeDeployConfig deployDir "module.nix"
+    "{ obeliskPkgs, ... }: {...}: {\n\
+    \  imports = [(obeliskPkgs.path + /nixos/modules/virtualisation/amazon-image.nix)];\n\
+    \  ec2.hvm = true;\n\
+    \}"
```
Comment (3noch) [inline lib/command/src/Obelisk/Command/Deploy.hs]: Would rather not inline this... somehow piggy back on obelisk thunk's version of this... (see `mkBaseEc2`)

**Comment by HectorAE** [comment]
This will address #444 right?

**Comment by 3noch** [comment]
Right. You'd have to configure it properly but it should at least be trivial after this.

**Comment by 3noch** [comment]
Tested both on EC2 and VirtualBox

**Comment by 3noch** [comment]
Documentation was added to README: https://github.com/obsidiansystems/obelisk/tree/0aaa9c828be84404e9513b56f3abef077b132f61#custom-non-ec2-deployment

**Pull request merged**